### DISCUSSION TITLE
Unmet Peer Dependencies Warning

### DISCUSSION THREAD

**Discussion by vedxp** [vuejs/vitepress#1409] [Q&A] [answered]
I get this warning while building:
<details><summary>Show</summary>

![Screenshot_20220928-095423~2](https://user-images.githubusercontent.com/87353286/192728132-879cd085-9349-4fd3-abb6-9b9562380010.png)

</details>

I tried this:

<details><summary>Show</summary>

![Screenshot_20220928-133646~2](https://user-images.githubusercontent.com/87353286/192728814-bb18a1ec-3cbc-4996-b313-7c0b8c341ee7.png)

</details>

Reference image how I added in my package.json:

<details><summary>Show</summary>

![Screenshot_20220928-135135~2](https://user-images.githubusercontent.com/87353286/192729913-72f89ff7-10d7-4c39-ab40-39d2038b7814.png)

</details>

I tried putting `pnpm` `npm` `yarn`, none seem to work.

Not sure what I'm doing wrong.
Help appreciated!

**Comment by brc-dd** [comment, marked as answer]
That will only work if you're using pnpm. Changing it to npm/yarn won't work. It's fine to just ignore that warning. Also, it just occurs on initial install. Looks like you haven't added your lock file to source control.

**Reply by vedxp** [reply]
> Looks like you haven't added your lock file to source control.

Yeah

Thanks for the answer!

**Reply by vedxp** [reply]
![Screenshot_20220928-160328~2](https://user-images.githubusercontent.com/87353286/192757791-0f4fcfea-f09b-4410-91c7-f7a9a0c38374.png)

What are these warnings about?

Is this something I did wrong?

@brc-dd 

Edit: This is showing after pushing yarn.lock to the repo.

**Reply by brc-dd** [reply]
Ignore them. Those aren't even warnings. Many packages are required only on a particular os and architecture (like fsevents is needed only on macOS) and many packages have prebuilt binaries depending on platform (like esbuild). In case of packages like fsevents, they are marked as optional deps as installing them on other platforms (like Linux in this case) doesn't make sense as they aren't required / won't work there. And in case of packages like esbuild, all of the prebuilt packages are marked as optional deps too. So, only those matching the current platform are installed (or built if no binary matching the current target is available) and rest are ignored (like in this case, installing android-arm version of esbuild isn't required on Linux).

**Reply by vedxp** [reply]
Ohk, got it.
Thanks for explaining!

I was just worried as these started appearing only after adding yarn.lock to the repo.

But, thanks again for the answer.
